stmhal/dma.c

```diff
@@ -33,9 +33,145 @@
 #include "py/obj.h"
 #include "irq.h"
 
-#define NSTREAMS_PER_CONTROLLER_LOG2 (3)
-#define NSTREAMS_PER_CONTROLLER (1 << NSTREAMS_PER_CONTROLLER_LOG2)
```

Comment:
As NSTREAMS_PER_CONTROLLER_LOG2 is 3 for F-series we have 8 streams. but on L4 there are only 7 streams. The use of this #define in dma_tickle is therfor no more possible. Therefore I removed NSTREAMS_PER_CONTROLLER_LOG2.